internal/clusterinfo/data.go

```diff
@@ -292,46 +304,159 @@ func (c *ClusterInfo) GetNSQDTopicProducers(topic string, nsqdHTTPAddrs []string
+			for _, t := range statsResp.Topics {
+				producerTopics = append(producerTopics, ProducerTopic{Topic: t.Name})
+			}
+
+			version, err := semver.Parse(infoResp.Version)
+			if err != nil {
+				version, _ = semver.Parse("0.0.0")
+			}
+
 			lock.Lock()
 			defer lock.Unlock()
+			success = true
+			producerList = append(producerList, &Producer{
+				Version:          infoResp.Version,
+				VersionObj:       version,
+				BroadcastAddress: infoResp.BroadcastAddress,
+				Hostname:         infoResp.Hostname,
+				HTTPPort:         infoResp.HTTPPort,
+				TCPPort:          infoResp.TCPPort,
+				Topics:           producerTopics,
+			})
+		}(addr)
+	}
+	wg.Wait()
+
+	if success == false {
+		return nil, errors.New("unable to query any nsqd")
+	}
+	return producerList, nil
+}
+
+// GetNSQDTopicProducers returns a ProducerList containing the addresses of all the nsqd
+// that produce the given topic
+func (c *ClusterInfo) GetNSQDTopicProducers(topic string, nsqdHTTPAddrs []string) (ProducerList, error) {
+	var success bool
+	var producerList ProducerList
+	var lock sync.Mutex
+	var wg sync.WaitGroup
+
+	type infoRespType struct {
+		Version          string `json:"version"`
+		BroadcastAddress string `json:"broadcast_address"`
+		Hostname         string `json:"hostname"`
+		HTTPPort         int    `json:"http_port"`
+		TCPPort          int    `json:"tcp_port"`
+	}
+
+	type statsRespType struct {
+		Topics []struct {
+			Name string `json:"topic_name"`
+		} `json:"topics"`
+	}
+
+	for _, addr := range nsqdHTTPAddrs {
+		wg.Add(1)
+		go func(addr string) {
+			defer wg.Done()
+
+			endpoint := fmt.Sprintf("http://%s/stats?format=json", addr)
+			c.logf("NSQD: querying %s", endpoint)
+
+			var statsResp statsRespType
+			err := http_api.NegotiateV1(endpoint, &statsResp)
 			if err != nil {
-				c.logf("ERROR: nsqd %s - %s", endpoint, err.Error())
+				c.logf("ERROR: nsqd %s - %s", endpoint, err)
 				return
 			}
+
+			var producerTopics ProducerTopics
+			for _, t := range statsResp.Topics {
+				producerTopics = append(producerTopics, ProducerTopic{Topic: t.Name})
+			}
+
+			lock.Lock()
 			success = true
+			lock.Unlock()
 
-			for _, t := range resp.Topics {
+			for _, t := range statsResp.Topics {
 				if t.Name == topic {
-					addresses = append(addresses, addr)
+					endpoint := fmt.Sprintf("http://%s/info", addr)
+					c.logf("NSQD: querying %s", endpoint)
+
+					var infoResp infoRespType
+					err := http_api.NegotiateV1(endpoint, &infoResp)
+					if err != nil {
+						c.logf("ERROR: nsqd %s - %s", endpoint, err)
+						return
+					}
+
+					version, err := semver.Parse(infoResp.Version)
+					if err != nil {
+						version, _ = semver.Parse("0.0.0")
+					}
+
+					lock.Lock()
+					defer lock.Unlock()
+					producerList = append(producerList, &Producer{
+						Version:          infoResp.Version,
+						VersionObj:       version,
+						BroadcastAddress: infoResp.BroadcastAddress,
+						Hostname:         infoResp.Hostname,
```

Comment:
It didn't occur to me at the time, but there is a compatibility issue with expecting this information back from the new `/info` endpoint. Namely w/o these fields nsqadmin is unable to display the topic or channel page for older hosts because it will attempt to query `/info` and use this empty field to query `/stats`.

```
[nsqadmin] 2015/09/21 13:37:11.922482 CI: querying nsqd http://127.0.0.1:4151/info
[nsqadmin] 2015/09/21 13:37:11.923098 CI: querying nsqd http://:0/stats?format=json
[nsqadmin] 2015/09/21 13:37:11.923515 ERROR: failed to get channel metadata - failed to query any nsqd
```